src/ocamlorg_package/lib/ocamlorg_package.ml

```diff
@@ -453,12 +455,15 @@ end = struct
     let atom = Re.alt [ name; author; tag; synopsis; description; plain ] in
     Re.compile atom
 
-  let to_request str =
+  let to_request ~nick_resolve str =
     let str = String.lowercase_ascii str in
     let to_constraint = function
       | [ _; s ] -> Any s
       | [ _; "tag:"; s ] -> Tag s
-      | [ _; "author:"; s ] -> Author s
+      | [ _; "author:"; s ] ->
+          Author
+            Option.(
+              value ~default:s (map String.lowercase_ascii (nick_resolve s)))
```

Comment:
Isn't it doing the same thing that was done before?

c.f. previous comment:
> Not sure that's correct. The search function in Ocamlorg_package will try to match the search with the author field so if you search by name and the author field contains the name, this introduces a regression.
> 
> Do we support multiple search? (e.g. author:"Thibaut Mattio" author:"tmattio"). If so, we can add the GitHub handle instead of replacing the search query.

Reply:
With the code above, my understanding is that if I type `author:"thibaut mattio"`, it will search for `author:tmattio`

Reply:
I've changed the names to reflect they work the other way around.

Reply:
The name/author/nick/email matching logic is now:
- Defined in `Data.Opam_user`
- Used in `Ocamlorg_package.search`
- Same as current online logic

Results with this PR should be the same as they used to be.

`Ocamlorg_package.Search` could be turned into a functor to avoid passing the `is_author_match` parameter from call to call.

Reply:
Cool! I suggest we move the matching logic out of Data though, we might change that to use fuzzy matching or something else in the future and how we search for authors shouldn't have any impact on data